### ISSUE TITLE
Aw Snap!

### ISSUE THREAD

**Issue by FSTABMurphy** [eerotal/LibreSignage#127]
Hi my name is Sam im new to github ive downloaded allsorts from here before but never actually made an account im here to describe a potential problem between libresignage and chromium in kiosk mode and i have a pi 4 running the libresignage display constantly but after a couple hours it aw snaps and dies the solution would be to have an auto refresh built into the page or a refresh the session on the client due to it running on wifi the latency can be measurably high if someone has a solution this would be my ticket out of them late nights scratching my head.
Thankyou for making this fantastic piece of software!!!

**Comment by eerotal** [comment]
That's odd. I wonder if the RPi is running out of memory somehow, although the newer generations have quite a lot of ram already. It'd be great if you could monitor the memory consumption of your browser until the Aw Snap! page appears so that I would be able to pinpoint the problem a bit better.

That said, I'm quite hesitant in implementing an automatic refresh feature to the web interface as that's really just a workaround and not a real solution to the problem. As another workaround you could of course just periodically restart the browser with a cron job for example.

**Comment by FSTABMurphy** [comment]
I have been monitoring the rpi4 for a while and the ram doesn't go above 500mb of ram when its got 2gb while its displaying it usually sits at 413mb drops to 405mb and slowly builds back to 413mb before dumping again so from my personal experience i dont think its ram ill include a screenshot of htop for you to see. i have also already created a cron for killing kiosk mode and relaunching the page but it will cut off the slide show mid way through to restart the browser im more than willing to give you any debug info you might need to fix this problem for the community thankyou so much for your support.

**Comment by FSTABMurphy** [comment]
![Screenshot](https://user-images.githubusercontent.com/66629545/84484098-31448c00-ac92-11ea-93ec-d4c64f3b5545.jpg)

**Comment by FSTABMurphy** [comment]
sorry for the quality

**Comment by eerotal** [comment]
Yeah, probably not memory related then. You could start chromium with logging enabled by running `chromium --enable-logging=stderr --v=1 > chromium_log.txt [LIBRESIGNAGE URL] 2>&1` from a terminal. Just replace `[LIBRESIGNAGE URL]` with the URL/IP of your LibreSignage instance. The log is written in chromium_log.txt in the directory where you run that command. I'm not yet sure how much relevant info I can get from that log but let's give it a try. It's probably only worth it to copy&paste some of the last lines of the log here because it'll probably be quite huge. Feel free to remove any personal info from the log if there happens to be any.

By the way, have you tried running LibreSignage with some other browser, like Firefox for example. I wonder if this is only a problem on Chromium.

**Comment by FSTABMurphy** [comment]
Ill implement this and get back when its "Awsnapped" because it takes time ive worked out it takes around 2 hours to awsnap but ive had it last for 10 hours before awsnap so ill get back when its done what it needs to
